### ISSUE TITLE
Feature suggestion : 3D Camera and 3D tracking

### ISSUE THREAD

**Issue by threedslider** [NatronGitHub/Natron#319]
It is a feature suggestion for low priority but interesting and powerful to make compo very realistic ! If someone developer or contributor can make this 3D camera and 3D tracking in Natron would be appreciated and also it brights to community more choice to make in VFX.

For reference  : [3D Viewing](http://web.cse.ohio-state.edu/~wang.3602/courses/cse5542-2013-spring/8-3D_viewing.pdf) and [2D/3D tracking](http://aims.robots.ox.ac.uk/wp-content/uploads/2016/01/AIMS-L5.pdf)

I will want to make this feature too if nobody implement it, but I am still learning to source code in my free time.

Thank you and Happy New year 2019 ! :)

**Comment by threedslider** [comment]
For those who want to know I am working first to 3D view and 3D camera requested by magdesign :)

**Comment by devernay** [comment]
This has been in the planned features for a long time, but I think the 2D compositing issues have to be fixed.
https://github.com/NatronGitHub/Natron#planned-features

Blender has a very good 3D tracking module, and I think it should be considered for all things 3D. 3D stuff in a video compositing app is often gimmicky anyway.

You can always try to work on it. As usual, fork Natron, work on it, and when it's done do a PR. This feature is probably a lot more work that what you expect it to be, though.

**Comment by threedslider** [comment]
Yes for 3D tracking in Blender is very good, I am learning from that too on how to make in Natron as inspired :)

I was forked again from Natron original in here, you know to update for RB-2.3 to specifically? I want to work on both master(3.0?) and RB-2.3 ?

Thanks to you devernay

**Comment by devernay** [comment]
Work on RB-2.3. master is extremely buggy and there is a chance that it gets dropped at some point.
I still merge any change to RB-2.3 into master, though

**Comment by threedslider** [comment]
Ok thank you I am working on RB-2.3 branch :)

**Comment by natowi** [comment]
Hi there, I just wanted to point you to the [AliceVision](https://github.com/alicevision/AliceVision) library (back-end of [Meshroom](https://github.com/alicevision/meshroom)) that could (also) be used for 3d reconstruction and camera tracking. (There are Blender and [Houdini](https://www.sidefx.com/tutorials/alicevision-plugin/) plugins, that can be used as reference)

**Comment by threedslider** [comment]
Thank you natowi for sharing this information, I will look into AliceVision from code.
